Automattic/jetpack
# Userless Connection: connection flow

#### Changes proposed in this Pull Request:
- Modify connection screen appearance
- Redirect "userless" users to the "Plans" page after they choose to proceed without connecting to WP.com
- `ConnectUserBar` React component
- New module setting: `requires_user_connection`, and new reducers to pull this data
- New property for the `SettingsGroup` component: `disableInUserlessMode`, which checks if the site is working in userless mode, and makes the `SettingsGroup` inactive
- Adjusting the modules that require user connection accordingly

#### Jetpack product discussion
p1HpG7-b0a-p2
p9dueE-2vd-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
__Free plan__
1. Define the `JETPACK_NO_USER_TEST_MODE`, disconnect Jetpack.
2. Initiate the connection using the In-Place flow.
3. Instead of clicking "Approve", scroll down to the "Or start using Jetpack now" section. Confirm that it looks like this:
<img width="614" alt="Screen Shot 2021-02-17 at 7 06 31 PM" src="https://user-images.githubusercontent.com/1341249/108289111-696e6980-7153-11eb-85e9-cdfa0b88bcb0.png">
4. Click "Continue without user account". You should be taken to the "Pricing" page: `https://cloud.jetpack.com/pricing/`.

5. Scroll down and click "Start for free". You should be redirected back to Jetpack Dashboard.


__Connecting user via In-Place Flow__
6. Go to "Jetpack -> Settings -> Security -> Downtime monitoring". Underneath it you should see the "Connect my user account" button:
<img width="891" alt="Screen Shot 2021-02-17 at 7 10 35 PM" src="https://user-images.githubusercontent.com/1341249/108289405-f7e2eb00-7153-11eb-8a8d-d7b8aecbf380.png">
7. Click "Connect my user account" button, the connection widget should unfold under the same "Downtime monitoring" card:
<img width="1054" alt="Screen Shot 2021-02-17 at 7 13 50 PM" src="https://user-images.githubusercontent.com/1341249/108289592-4abca280-7154-11eb-903f-d42642d2bdd2.png">
8. Click "Approve". After the authorization has completed, the connection widget should get replaced with the "Configure your notification settings" link in the same place (under "Downtime monitoring"). This should happen **without** page reloading.

__Purchasing a plan__
_You need to disconnect from the Automattic Proxy to use the production build of Jetpack Cloud. This flow does not fully work with the Jetpack Cloud Stage, which is served to you when you connect through the proxy._

Your WP.com account may be in 3 different states in regard to this account, please repeats testing instructions in this section in each state:
- Not logged into WP.com
- Logged into WP.com, not connected to Jetpack Cloud app (go to the [Connected Apps](https://wordpress.com/me/security/connected-applications) section and disconnect all "Jetpack.com" apps if there are any).
- Logged into WP.com, connected to Jetpack Cloud app (go to https://cloud.jetpack.com/ and approve the connection).

1. Disconnect Jetpack, and click "Set up Jetpack" again.
2. Click "Continue without user account". You should be taken to the "Pricing" page: `https://cloud.jetpack.com/pricing/`
3. Choose a plan and click on the Purchase button. You should be redirected to the Calypso flow to connect your user account to WP.com. If you are not logged into WP.com, the flow will include authentication process.
4. After you click "Approve" you should get redirected to Checkout.
5. _This step may only be completed once, no need to repeat it for every WP.com account state_. Complete the checkout and confirm that the product is purchased successfully.

#### Proposed changelog entry for your changes:
* Using Jetpack without connecting to WP.com account.
